extensions/ql-vscode/src/run-queries.ts

```diff
@@ -199,15 +199,21 @@ export class QueryEvaluationInfo {
         });
         if (await this.hasEvalLog()) {
           queryInfo.evalLogLocation = this.evalLogPath;
-          await qs.cliServer.generateLogSummary(this.evalLogPath, this.evalLogSummaryPath, this.evalLogEndSummaryPath);
-          queryInfo.evalLogSummaryLocation = this.evalLogSummaryPath;
-          fs.readFile(this.evalLogEndSummaryPath, (err, buffer) => {
-            if (err) {
-              throw new Error(`Could not read structured evaluator log end of summary file at ${this.evalLogEndSummaryPath}.`);
-            }
-            void qs.logger.log(' --- Evaluator Log Summary --- ');
-            void qs.logger.log(buffer.toString());
-          });
+          void qs.cliServer.generateLogSummary(this.evalLogPath, this.evalLogSummaryPath, this.evalLogEndSummaryPath)
+            .then(() => {
+              queryInfo.evalLogSummaryLocation = this.evalLogSummaryPath;
+              fs.readFile(this.evalLogEndSummaryPath, (err, buffer) => {
+                if (err) {
+                 throw new Error(`Could not read structured evaluator log end of summary file at ${this.evalLogEndSummaryPath}.`);
+                }
+                void qs.logger.log(' --- Evaluator Log Summary --- ');
+                void qs.logger.log(buffer.toString());
+              });
+            })
+
+            .catch(err => {
+              void logger.log(`Failed to generate structured evaluator log summary. Reason: ${err.message}`);
```

Comment:
Should we use `showAndLogWarningMessage` here so that the error is displayed in a pop-up as well as being logged?

Reply:
Hm, I think because all the other related popups in this realm are warning messages I'll keep it as a warning message.

Reply:
I think I'm inclined to agree that everything in this code should be a warning since failures here aren't fatal in the sense they don't result in a failure to produce results for the query which is what most users care about.